org.lflang/src/org/lflang/cli/Lfc.java

```diff
@@ -52,83 +56,98 @@ public Lfc() {
-            return opts;
-        }
+    @Option(
+        names = "--build-type",
+        description = "The build type to use.")
+    private String buildType;
 
-        /**
-         * Return a list of options that are to be passed on to the code
-         * generator.
-         *
-         * @return List of options that must be passed on to the code gen stage.
-         */
-        public static List<Option> getPassedOptions() {
-            return Arrays.stream(CLIOption.values())
-                .filter(opt -> opt.passOn).map(opt -> opt.option)
-                .collect(Collectors.toList());
-        }
-    }
+    @Option(
+        names = {"-c", "--clean"},
+        arity = "0",
+        description = "Clean before building.")
+    private boolean clean;
+
+    @Option(
+        names = "--target-compiler",
+        description = "Target compiler to invoke.")
+    private String targetCompiler;
+
+    @Option(
+        names = "--external-runtime-path",
+        description = "Specify an external runtime library to be used by the"
+                    + " compiled binary.")
+    private String externalRuntimePath;
+
+    @Option(
+        names = {"-f", "--federated"},
+        arity = "0",
+        description = "Treat main reactor as federated.")
+    private boolean federated;
+
+    @Option(
+        names = "--logging",
+        description = "The logging level to use by the generated binary")
+    private String logging;
+
+    @Option(
+        names = {"-l", "--lint"},
+        arity = "0",
+        description = "Enable or disable linting of generated code.")
+    private boolean lint;
+
+    @Option(
+        names = {"-n", "--no-compile"},
+        arity = "0",
+        description = "Do not invoke target compiler.")
+    private boolean noCompile;
+
+    @Option(
+        names = {"-o", "--output-path"},
+        defaultValue = "",
+        fallbackValue = "",
+        description = "Specify the root output directory.")
+    private String outputPath;
+
+    @Option(
+        names = {"-q", "--quiet"},
+        arity = "0",
+        description = 
+            "Suppress output of the target compiler and other commands")
+    private boolean quiet;
+
+    @Option(
+        names = {"-r", "--rti"},
+        description = "Specify the location of the RTI.")
+    private String rti;
+
+    @Option(
+        names = "--runtime-version",
+        description = "Specify the version of the runtime library used for"
+                    + " compiling LF programs.")
+    private String runtimeVersion;
+
+    @Option(
+        names = {"-s", "--scheduler"},
+        description = "Specify the runtime scheduler (if supported).")
+    private String scheduler;
+
+    @Option(
+        names = {"-t", "--threading"},
+        arity = "0",
+        description = "Specify whether the runtime should use multi-threading"
+                    + " (true/false).")
+    private boolean threading;
+
+    // TODO: could be int?
```

Comment:
Ye, I think this should be processed as an integer

Reply:
As mentioned above - I realized it gets converted back to a string anyway, so I deleted the comment and left it as is.